### PR TITLE
supermodel: init at 0-unstable-2024-04-05

### LINKED ISSUE #302099: Package request: supermodel
**Project description**

Supermodel emulates Sega's Model 3 arcade platform, allowing you to relive state-of-the-art 3D arcade gaming as it existed from 1996 through 1999.

**Metadata**

* homepage URL: https://supermodel3.com/
* source URL: https://github.com/trzy/Supermodel
* license: GPL
* platforms: linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by msanft**
## Description of changes

Add the [Supermodel](https://github.com/trzy/Supermodel) SEGA Model 3 emulator.

image

For testing, you will need to:
- Copy / Link the configuration file to your home: `cp result/share/supermodel/Config/Supermodel.ini ~/.config/supermodel/Config/Supermodel.ini`
- Copy / Link the assets to your home: `cp -r result/share/supermodel/Assets/* ~/.local/share/supermodel/Assets/`

Then the emulator can be started with `./result/bin/supermodel -game-xml-file=result/share/supermodel/Config/Games.xml /path/to/romset`. 
While I would normally prefer to package the application to work out of the box, it unfortunately doesn't allow enough configuration to do so. (e.g. specifying a config path) Patches were not feasible for the amount of code that'd be needed.

Closes #302099 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 302417` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>supermodel</li>
  </ul>
</details>

**Comment by keenanweaver** [comment]
image
Working great after following the instructions. Bummer it can't work out of the box. Setting a simple home.file with home-manager is not a big deal for me, but for others that won't be the case. Something to take upstream? Otherwise, definitely something to put in the longDescription.

**Comment by msanft** [comment]
> Something to take upstream?

Might be. The project seems to be actively maintained. I'll put some usage guidelines into `meta.longDescription` and see if we can get some more configurability upstream.

**Review by keenanweaver — APPROVED** [review]

**Review by katexochen — APPROVED** [review]
lgtm

File: pkgs/by-name/su/supermodel/package.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    longDescription = ''
+      Supermodel requires specific files to be present in the $HOME directory of
+      the user running the emulator. To ensure these files are present, move the
+      configuration and assets as follows:
+
+      <code>cp $out/share/supermodel/Config/Supermodel.ini ~/.config/supermodel/Config/Supermodel.ini</code>
+      <code>cp -r $out/share/supermodel/Assets/* ~/.local/share/supermodel/Assets/</code>
+
+      Then the emulator can be started with:
+      <code>supermodel -game-xml-file=$out/share/supermodel/Config/Games.xml /path/to/romset</code>.
```
Comment (katexochen) [inline pkgs/by-name/su/supermodel/package.nix:58]: Could the `-game-xml-file` flag be added as a wrapper? Looks like it is always required, and point into the derivation, so the path is already known at build time.

Comment (msanft) [reply pkgs/by-name/su/supermodel/package.nix:58]: The file passed here is an examplary file from Supermodel, which I ship within the closure for convenience. Some games may require custom files, which is why I'd vouch against wrapping it

Comment (katexochen) [reply pkgs/by-name/su/supermodel/package.nix:58]: Got it, thanks. :)

**Pull request merged**